imports/blocks/give/index.js

```diff
@@ -183,118 +159,51 @@ export default class Give extends Component {
     this.props.dispatch(modal.render(OnBoard, props));
   }
 
-  render() {
-    const {
-      data,
-      url,
-      errors,
-      step,
-      transactions,
-      schedules,
-      total,
-      savedAccount,
-      state,
-      transactionType,
-      scheduleToRecover,
-    } = this.props.give;
-
-    let { campuses, states, countries } = this.props.data;
-
-    if (!campuses) {
-      campuses = defaultArray;
-    }
+  save = (...args) => {
+    this.props.dispatch(giveActions.save(...args));
+  }
 
-    campuses = campuses.map((x) => ({ label: x.name, value: x.id }));
+  clear = (...args) => {
+    this.props.dispatch(giveActions.clear(...args));
+  }
 
-    if (!states) {
-      states = defaultArray;
-    }
+  clearData = () => {
+    this.props.dispatch(giveActions.clearData());
+    this.props.dispatch(modal.hide());
+  }
 
-    states = states.map((x) => ({ label: x.name, value: x.value }));
+  render() {
+    let {
+      campuses = defaultArray,
```

Comment:
this would go away as defaults would be passed from `props` function of `graphql`